mbuild/conversion.py

```diff
@@ -0,0 +1,1364 @@
+    if relative_to_module:
+        script_path = os.path.realpath(
+            sys.modules[relative_to_module].__file__)
+        file_dir = os.path.dirname(script_path)
+        filename = os.path.join(file_dir, filename)
+
+    # Handle the case of a xyz and json file, which must use an internal reader
+    extension = os.path.splitext(filename)[-1]
+
+    if not backend:
+        try:
+            # Try matching backend based on extension
+            backend = default_backends[extension]
+        except KeyError:
+            # Else use default backend
+            backend = 'mdtraj'
+
+    # First check internal readers
+    if backend == 'internal':
+        # Handle json format
+        if extension == '.json':
+            # This doesn't seem to handle the case when compound is given
+            compound = compound_from_json(filename)
+            return compound
+        # Handle xyz file
+        if extension == '.xyz' and not 'top' in kwargs:
+            if coords_only:
+                tmp = read_xyz(filename)
+                if tmp.n_particles != compound.n_particles:
+                    raise ValueError('Number of atoms in {filename}'
+                        'does not match {compound}'.format(**locals()))
+                ref_and_compound = zip(tmp._particles(include_ports=False),
+                                       compound.particles(include_ports=False))
+                for ref_particle, particle in ref_and_compound:
+                    particle.pos = ref_particle.pos
+            else:
+                compound = read_xyz(filename, compound=compound)
+        elif extension == '.xyz' and 'top' in kwargs:
+            backend = 'mdtraj'
+
+    # Then pybel reader
+    elif backend == 'pybel':
+        pybel = import_('pybel')
+        if extension == '.sdf':
+            pybel_mol = pybel.readfile('sdf', filename)
+            # pybel returns a generator, so we grab the first molecule of a
+            # list of len 1.
+            # Raise ValueError if there are more molecules
+            pybel_mol = [i for i in pybel_mol]
+            if len(pybel_mol) == 1:
+                compound = from_pybel(
+                            pybel_mol=pybel_mol[0],
+                            compound=compound,
+                            coords_only=coords_only,
+                            infer_hierarchy=infer_hierarchy)
+            else:
+                raise ValueError('More tahn one pybel molecule in file,'
+                            'more than one pybel molecule is not supported')
+        # text file detected, asssume contain smiles string
+        elif extension == '.txt':
+            # Fail-safe measure
+            compound = load_smiles(filename, compound)
+
+    # Then parmed reader
+    elif backend == 'parmed':
+        warn('Using parmed reader. Bonds may be inferred '
+             'from inter-particle distances and standard '
+             'residue templates')
```

Comment:
I like this warning. I might add "Please check that the bonds in the mbuild.Compound are correct."